### ISSUE TITLE
getPositionStream not working for ios

### ISSUE THREAD

**Issue by andreamr97** [Baseflow/flutter-geolocator#1076]
## 🐛 Bug Report

getPositionStream only working on Android not on iOS

when changing location on Android simulator it works no problem, but on iOS it crashes 

![image](https://user-images.githubusercontent.com/101415962/172956457-df71c298-1028-493d-a237-28b51092920c.png)

### Configuration

**Version:** ^8.2.1

**Platform:** 
- [15.5 ] :iphone: iOS
- [ ] :robot: Android

**Comment by andreamr97** [comment]
this is my code: 

class _MMNewMerchantList extends State<MMNewMerchantList> {
  Position? _currentUserPosition;
  late bool serviceEnabled;
  late LocationPermission permission;
  late Position valueFinal;

  Future<Placemark> _determinePosition() async {
    serviceEnabled = await Geolocator.isLocationServiceEnabled();
    if (!serviceEnabled) {
      return Future.error('Location services are disabled.');
    }

    permission = await Geolocator.checkPermission();
    if (permission == LocationPermission.denied) {
      permission = await Geolocator.requestPermission();
      if (permission == LocationPermission.denied) {
        return Future.error('Location permissions are denied');
      }
    }

    if (permission == LocationPermission.deniedForever) {
      return Future.error(
          'Location permissions are permanently denied, we cannot request permissions.');
    }
    _currentUserPosition = await Geolocator.getCurrentPosition(desiredAccuracy: LocationAccuracy.high);
    print("current position: $_currentUserPosition");

    List<Placemark> placemarks = await placemarkFromCoordinates(_currentUserPosition!.latitude, _currentUserPosition!.longitude);
    Placemark place = placemarks[0];
    print(place.locality);
    return place;
  }

  @override
  void initState() {
    super.initState();
    final model = GetIt.I<MMDiscoverViewModel>();
    model.globalContext = context;

      final LocationSettings locationSettings = LocationSettings(
        accuracy: LocationAccuracy.high,
       distanceFilter: 100,
      );
     Geolocator.getPositionStream(locationSettings: locationSettings).listen(
            (Position? position) {
              _determinePosition().then((value) {
                GetIt.I<MMDiscoverViewModel>().getAllMerchantsFilter(value);
                GetIt.I<MMDiscoverViewModel>().currentPosition = value;
              });
        });

  }

**Comment by mvanbeusekom** [comment]
Hi @andreamr97,

The error you receive indicates that the user hasn't provided permissions to access location information on that device. Looking at the provided code sample this makes sense as it doesn't request permissions before starting listening to the position stream.

Another issue I notice is that the code receives a position from the stream but ignores it, calls `_determinePosition` and uses `getCurrentPosition` requesting a new position. This is a bit of a waste on resources (as you already have the position) and specifically on iOS this causes a problem as `getCurrentPosition` will close the stream (unfortunately this is how Apple implemented the location services).

A better solution would be:

1. Check if location services are enabled;
2. Check and request location permissions;
3. Start listening for position updates;
4. When a position arrives, feed it to the `placeMarkFromCoordinates` method.

In code this could look something like this:

```dart
class _MMNewMerchantList extends State<MMNewMerchantList> {
  Position? _currentUserPosition;
  late bool serviceEnabled;
  late LocationPermission permission;
  late Position valueFinal;

  Future<void> _handlePermissions() {
    serviceEnabled = await Geolocator.isLocationServiceEnabled();
    if (!serviceEnabled) {
      return Future.error('Location services are disabled.');
    }

    permission = await Geolocator.checkPermission();
    if (permission == LocationPermission.denied) {
      permission = await Geolocator.requestPermission();
      if (permission == LocationPermission.denied) {
        return Future.error('Location permissions are denied');
      }
    }

    if (permission == LocationPermission.deniedForever) {
      return Future.error(
          'Location permissions are permanently denied, we cannot request permissions.');
    }
  } 

  void _listenForPositionUpdates() {
      final LocationSettings locationSettings = LocationSettings(
        accuracy: LocationAccuracy.high,
       distanceFilter: 100,
      );
      
     Geolocator.getPositionStream(locationSettings: locationSettings).listen(
            (Position? position) {
              if (position != null) {
                _determinePlacemark(position).then((Placemark placemark) {
                  GetIt.I<MMDiscoverViewModel>().getAllMerchantsFilter(value);
                  GetIt.I<MMDiscoverViewModel>().currentPosition = value;
                }); 
              }
              _determinePosition().then((value) {
                GetIt.I<MMDiscoverViewModel>().getAllMerchantsFilter(value);
                GetIt.I<MMDiscoverViewModel>().currentPosition = value;
              });
        });
  }

  Future<Placemark> _determinePlacemark(Position position) async {
    List<Placemark> placemarks = await placemarkFromCoordinates(position.latitude, position.longitude);
    Placemark place = placemarks[0];
    print(place.locality);
    return place;
  }

  @override
  void initState() {
    super.initState();
    final model = GetIt.I<MMDiscoverViewModel>();
    model.globalContext = context;

    _handlePermissions().then(() {
      _listenForPositionUpdates();
    }).catchError((e) {
       // TODO: Handle permission error...
    });
  }
```

I have quickly put this code together to give you an idea on a possible solution. Since I don't have all dependencies I wasn't able to compile and validate it but it should give you some understanding on where I am going with this.

For now I will close this issue as this seems to be a user error and not a problem with the plugin itself.. Don't hesitate to leave a comment if you have followup questions.

**Comment by andreamr97** [comment]
ahhh im so embarrassed lol thank you very much!!!
